Cloth.Persistence.Ef/Repositories/OrderRepository.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Cloth.Application.Extensions;
+using Cloth.Application.Interfaces;
+using Cloth.Domain.Entities;
+using Cloth.Domain.Exceptions;
+using Cloth.Persistence.Ef.Context;
+using Microsoft.EntityFrameworkCore;
+using Persistence.Abstractions.Repositories;
+
+namespace Cloth.Persistence.Ef.Repositories;
+
+public class OrderRepository : GenericRepository<Order>, IOrderRepository
+{
+    protected readonly ClothInventoryDbContext _dbContext;
+    public OrderRepository(ClothInventoryDbContext dbContext) : base(dbContext)
+    {
+        _dbContext = dbContext;
+    }
+
+    public async Task<IEnumerable<Order>> FilterOrdersAsync(DateTime? minDate, DateTime? maxDate, Guid? userId, Guid? statusId)
+    {
+        var allItems = await _dbContext.Orders
+            .Include(o => o.User)
+            .Include(o => o.Status)
+            .Where(p => p.IsDeleted == false)
+            .ToListAsync();
+
+        if (allItems == null || allItems.Count == 0)
+        {
+            throw new ItemNotFoundException("No items found in the database.");
+        }
+
+        allItems = allItems.OrderUserFilter(userId)
+            .OrderStatusFilter(statusId)
+            .MinDateFilter(minDate)
+            .MaxDateFilter(maxDate);
+
+        return allItems;
+    }
+
+    public async Task<IEnumerable<Order>> GetAllOrders()
+    {
+        var result = await _dbContext.Orders
+             .Include(o => o.User)
+             .Include(o => o.Status)
+             .Where(p => p.IsDeleted == false)
+             .ToListAsync();
+
+        return result;
+    }
+
+    public async Task<Order> GetOrderById(Guid orderId)
+    {
+        var result = await _dbContext.Orders
+             .Include(o => o.User)
+             .Include(o => o.Status)
+             .Include(o => o.OrderLines)
+                 .ThenInclude(od => od.Cloth)
+                 .ThenInclude(p => p.ClothSizes)
+                 .ThenInclude(cs => cs.Size)
+             .Where(p => p.IsDeleted == false)
+             .FirstOrDefaultAsync(o => o.Id == orderId);
+
+        if (result == null)
+        {
+            throw new ItemNotFoundException($"Order: {orderId} not found.");
```

Comment:
Same as  other similar case:

For me this is more optimized code:
```
var result = await _dbContext.Orders
    .AsNoTracking()
    .Include(o => o.User)
    .Include(o => o.Status)
    .Include(o => o.OrderLines)
        .ThenInclude(od => od.Cloth)
            .ThenInclude(p => p.ClothSizes)
                .ThenInclude(cs => cs.Size)
    .SingleAsync(o => o.Id == orderId && !o.IsDeleted);
```
  .AsNoTracking() // If you don't need to track changes
  https://learn.microsoft.com/en-us/ef/core/querying/tracking